pkg/api/v1/pod/util.go

```diff
@@ -313,6 +313,13 @@ func IsPodTerminal(pod *v1.Pod) bool {
 	return IsPodPhaseTerminal(pod.Status.Phase)
 }
 
+// IsPodDone returns true if it is certain that none of the containers are running and never will run.
+func IsPodDone(pod *v1.Pod) bool {
```

Comment:
the distinction between IsPodDone and IsPodTerminal is going to be pretty confusing for callers... I expect people to choose incorrectly... when should someone use IsPodTerminal instead of IsPodDone?

Reply:
I am not sure. Perhaps all users of `IsPodTerminal` should also check for the additional condition in `IsPodDone`? Then we don't need two functions. Let's see... there are exactly two calls outside of tests:

- https://github.com/kubernetes/kubernetes/blob/51429cb5afad998cae107e1917b88539fa95bdb8/pkg/controller/job/job_controller.go#L1008
- https://github.com/kubernetes/kubernetes/blob/51429cb5afad998cae107e1917b88539fa95bdb8/pkg/controller/util/endpoint/controller_utils.go#L99

The first looks like it would be okay with the semantic from `IsPodDone`. The second has explicit handling of `DeletionTimestamp`, so it's probably not okay to change it.

Let's defer this. I'll make `IsPodDone` local to `pkg/controller/resourceclaim/controller.go` for now and unless someone cares enough, will probably simply keep it there.

/cc @alculquicondor - you suggested moving this to `pkg/api/v1/pod/util.go` earlier.

Reply:
See https://github.com/kubernetes/kubernetes/compare/e68f74fc3978920769a4791ef4fb7aca5a426e7f..a514f40131c298ef7ca375e5ab8fa8ddafa3e74d

Reply:
@liggit: okay to move ahead without the pkg/api/v1/pod change?

Then please approve the auth change.

Reply:
@liggitt : okay to move ahead without the pkg/api/v1/pod change?

If yes, then please approve the auth change.